src/acl_mem.cpp

```diff
@@ -2060,8 +2060,8 @@ CL_API_ENTRY cl_int CL_API_CALL clEnqueueReadImageIntelFPGA(
         image->fields.image_objs.image_desc->image_width * src_element_size;
   }
 
-  if (image->mem_object_type == CL_MEM_OBJECT_IMAGE1D &&
-      image->mem_object_type == CL_MEM_OBJECT_IMAGE1D_ARRAY &&
+  if (image->mem_object_type == CL_MEM_OBJECT_IMAGE1D ||
+      image->mem_object_type == CL_MEM_OBJECT_IMAGE1D_ARRAY ||
       image->mem_object_type == CL_MEM_OBJECT_IMAGE1D_BUFFER) {
```

Comment:
Do we implement this feature now then XD? Or we can remove this condition and only keep what's under the `else` block...

Reply:
It's no where to be found. `image->fields.image_objs.image_desc->image_height` seems to be never initialized by default to `1` if the mem object is a 1D image. It is never initialized at all in fact. There's no unit test that tests this case as well. I may have to write some checks on these features.

Reply:
To be fair, we could also just initialize the variables on declaration with a default value. But it could be that the original intention was to avoid that.